src/opm/output/eclipse/AggregateGroupData.cpp

```diff
@@ -168,87 +110,44 @@ allocate(const std::vector<int>& inteHead)
-        //The field group always has seqIndex = 0 because it is always defined first
-        //Hence the all groups except the Field group uses the seqIndex assigned
-        //iGrp contains the child groups in ascending group sequence index
-        std::vector<size_t> childGroupIndex;
-        Opm::RestartIO::Helpers::groupMaps groupMap;
-        groupMap.currentGrpTreeNameSeqIndMap(sched, simStep, groupMapNameIndex,mapIndexGroup);
-        const auto indGroupMap = groupMap.indexGroupMap();
-        const auto gNameIndMap = groupMap.groupNameIndexMap();
-        for (auto* grp : childGroups) {
-            auto groupName = grp->name();
-            auto searchGTName = gNameIndMap.find(groupName);
-            if (searchGTName != gNameIndMap.end()) {
-                childGroupIndex.push_back(searchGTName->second);
-            }
-            else {
-                throw std::invalid_argument( "Invalid group name" );
-            }
-        }
-        std::sort(childGroupIndex.begin(), childGroupIndex.end());
-        for (auto groupTreeIndex : childGroupIndex) {
-            auto searchGTIterator = indGroupMap.find(groupTreeIndex);
-            if (searchGTIterator != indGroupMap.end()) {
-                auto gname = (searchGTIterator->second)->name();
-                auto gSeqNoItr = groupMapNameIndex.find(gname);
-                if (gSeqNoItr != groupMapNameIndex.end()) {
-                    iGrp[igrpCount] = (gSeqNoItr->second) + 1;
-                    igrpCount+=1;
-                }
-                else {
-                    std::cout << "AggregateGroupData, ChildGroups - Group name: groupMapNameIndex: " << gSeqNoItr->first << std::endl;
-                    throw std::invalid_argument( "Invalid group name" );
-                }
-            }
-            else {
-                std::cout << "AggregateGroupData, ChildGroups - GroupTreeIndex: " << groupTreeIndex << std::endl;
-                throw std::invalid_argument( "Invalid GroupTree index" );
-            }
+    } else if (group.groups().size() != 0) {
+        int igrpCount = 0;
+        for (const auto& group_name : group.groups()) {
+            const auto& child_group = sched.getGroup2(group_name, simStep);
+            iGrp[igrpCount] = child_group.insert_index();
+            igrpCount += 1;
         }
     }
 
     //assign the number of child wells or child groups to
     // location nwgmax
-    iGrp[nwgmax] =  (childGroups.size() == 0)
-        ? childWells.size() : childGroups.size();
-
-    // find the group type (well group (type 0) or node group (type 1) and store the type in
-    // location nwgmax + 26
-    const auto grpType = groupType(sched, group, simStep);
-    iGrp[nwgmax+26] = grpType;
+    if (group.wellgroup()) {
+        iGrp[nwgmax] = group.wells().size();
+        iGrp[nwgmax + 26] = 0;  // Well group: 0
+    } else {
+        iGrp[nwgmax] = group.groups().size();
+        iGrp[nwgmax + 26] = 1;  // node group: 1
+    }
 
```

Comment:
If we keep the `groupType()` function then this would be more easily written as
```C++
iGrp[nwgmax] = group.wellgroup() ? group.wells().size() : group.groups().size();
iGrp[nwgmax + 26] = groupType(group);
```
That said, it might actually be even more readable if the group size conditional was put into a helper function too.

Reply:
OK - I will implement the `groupSize()` helper.